app/routes/goal.py

```diff
@@ -0,0 +1,106 @@
+from flask import Blueprint, jsonify, request
+from app import db
+from app.models.goal import Goal
+from .routes_helper import validate_model
+from app.models.task import Task
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+    
+    new_goal = Goal.from_dict(request_body)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    goal_dict = Goal.to_dict(new_goal)
+
+    return jsonify({"goal": goal_dict}), 201
+
+
+@goals_bp.route("", methods=["GET"])
+def get_all_goals():
+    goals = Goal.query.all()
+
+    response = []
+
+    for goal in goals:
+        goal_dict = {
+            "id": goal.goal_id,
+            "title": goal.title
+        }
+        response.append(goal_dict)
+    
+    return jsonify(response), 200
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    selected_goal = validate_model(Goal, goal_id)
+
+    goal_dict = Goal.to_dict(selected_goal)
+
+    return jsonify({"goal": goal_dict}), 200
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def update_goal(goal_id):
+    selected_goal = validate_model(Goal, goal_id)
+
+    request_body = request.get_json()
+
+    if "title" not in request_body:
+        return jsonify({"message": "Request must include title and description"}), 400
```

Comment:
You're checking only the title here, but you tell them that title and description are required.

Reply:
Thank you for pointing that out!